common/define-grammar.js

```diff
@@ -627,11 +627,14 @@ module.exports = function defineGrammar(dialect) {
       type_annotation: $ => seq(':', $._type),
 
       asserts: $ => seq(
-        ':',
         'asserts',
         choice($.type_predicate, $.identifier, $.this)
       ),
 
+      asserts_annotation: $ => seq(
```

Comment:
Maybe rename to assertion_signature?

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html ctrl-f for assertion signature for reference

Reply:
I used `asserts_annotation` by analogy with `type_annotation` and `type_predicate_annotation`, compare lines 888 and 960. If we rename it to `assertion_signature`, I feel like we should also rename at least `type_predicate_annotation` to `type_predicate_signature`.

Reply:
Yeah that's fine, we can leave it as is for consistency